### PR TITLE
feat(git)!: add `force-if-includes` flag

### PR DESCRIPTION
**Pull request by UnnitMetaliya**
## Standards checklist:

- [x] The PR title is descriptive.
- [x] The PR doesn't replicate another PR which is already open.
- [x] I have read the contribution guide and followed all the instructions.
- [x] The code follows the code style guide detailed in the wiki.
- [x] The code is mine or it's from somewhere with an MIT-compatible license.
- [x] The code is efficient, to the best of my ability, and does not waste computer resources.
- [x] The code is stable and I have tested it myself, to the best of my abilities.
- [x] If the code introduces new aliases, I provide a valid use case for all plugin users down below.

## Changes:

- As per the issue #11388 adds a new safer flag in git plugin `--force-if-includes`

### CODE REVIEW

**Comment by carlosala** [comment]
Thanks for your contribution!

File: plugins/git/README.md [hunk 1]
```diff
@@ -120,6 +121,8 @@ plugins=(... git)
 | gp                   | git push                                                                                                                                                                                 |
 | gpd                  | git push --dry-run                                                                                                                                                                       |
 | gpf                  | git push --force-with-lease                                                                                                                                                              |
```
Comment (tom-reinders) [inline plugins/git/README.md:123]: Should have replaced this line, not added to

Comment (carlosala) [reply plugins/git/README.md:123]: Fixed, thanks for the review!

**Comment by aekt** [comment]
@UnnitMetaliya @carlosala thanks for contributions, but seems `--force-if-includes` having a different behaviour than the original `gpf`, can we consider keeping both? Thanks

**Comment by carlosala** [comment]
@aekt yes, we decided on purpose to make the check stricter, it's flagged as a breaking change. What are you missing from previous behaviour?

**Comment by aekt** [comment]
Thanks for quick reply @carlosala, probably because I don't a lot of branches locally...

I'm agree with the purpose, but do you think we can introduce a new one that is less strict? Say `gpfl` or `gpfrc`?

**Comment by carlosala** [comment]
let's go for `gpfl`!

**Comment by mcornella** [comment]
Can you describe the behavior it has on your case and why it's necessary to keep the old one?

**Comment by aekt** [comment]
As a minimal example, I have a commit graph like this:

```
* e97c34b (HEAD -> main) Update amend
| * f229734 (origin/main-tmp) Update
|/  
* 7aaf892 (origin/main, origin/HEAD) Initial commit
```

Then when I `gpf origin main:main-tmp` or `gpf origin HEAD:main-tmp` (this is what I mostly do), it warns me as expected:

```
To github.com:aekt/gpf.git
 ! [rejected]        main -> main-tmp (remote ref updated since checkout)
error: failed to push some refs to 'github.com:aekt/gpf.git'
hint: Updates were rejected because the tip of the remote-tracking
hint: branch has been updated since the last checkout. You may want
hint: to integrate those changes locally (e.g., 'git pull ...')
hint: before forcing an update.
```

**Comment by rwmitchell** [comment]
Everyone is certainly capable of creating their own aliases to make commands they use often, easier.

Is this a common workflow shared by most people?

**Comment by mcornella** [comment]
That will be solved by `--force` which bypasses all checks and will do whatever you ask it. This is already in the alias `gpf!`.

**Comment by rwmitchell** [comment]
How does adding an exclamation actually help?  If I mistype and accidentally hit ```gcn[TAB]``` the exclamation gets automatically added and if the tab is assumed to have done the right thing because it completely expanded something, it gets done.

Is there a way to block an exclamation from ever being expanded and required to be appended manually?

**Pull request merged**